modules/gateway/peers.go

```diff
@@ -257,31 +299,33 @@ func (g *Gateway) Connect(addr modules.NetAddress) error {
 		},
 		sess: muxado.Client(conn),
 	})
-	g.mu.Unlock(id)
+	g.mu.Unlock()
 
 	// call initRPCs
-	id = g.mu.RLock()
+	g.mu.RLock()
 	for name, fn := range g.initRPCs {
-		go g.RPC(addr, name, fn)
+		go func(name string, fn modules.RPCFunc) {
+			g.RPC(addr, name, fn)
+		}(name, fn)
```

Comment:
Correct. I kept the function body because the next PR will introduce error logging for the RPC and keeping the body will make the merge easier (the `gateway-logging` PR originally made on the `v061` branch)